src/DistributedTracingManager.ts

```diff
@@ -0,0 +1,241 @@
+                }
+            });
+        }
+
+        if (updateType !== DistributedSettingUpdateType.OnlySamplingRate) {
+            let selectedItem: SamplingModeItem = await vscode.window.showQuickPick(
+                this.getSamplingModePickupItems(),
+                { placeHolder: "Select whether to enable/disable the distributed tracing...", ignoreFocusOut: true },
+            );
+            if (!selectedItem) {
+                return;
+            }
+            mode = selectedItem.distributedTracingEnabled;
+        }
+
+        if (updateType !== DistributedSettingUpdateType.OnlyMode) {
+            if (mode !== false) {
+                samplingRate = await this.promptForSamplingRate(`Enter sampling rate, integer within [0, 100]`, samplingRate);
+
+                if (samplingRate === undefined) {
+                    return;
+                }
+            }
+        }
+
+        await vscode.window.withProgress({
+            title: `Update Distributed Tracing Setting`,
+            location: vscode.ProgressLocation.Notification,
+        }, async () => {
+            try {
+                const result = await this.updateDeviceTwin(mode, samplingRate, iotHubConnectionString, deviceIds);
+                TelemetryClient.sendEvent(Constants.IoTHubAIUpdateDistributedSettingDoneEvent,
+                    { Result: "Success", UpdateType: updateType.toString(), DeviceCount: deviceIds.length.toString(),
+                    SamplingRate: samplingRate ? samplingRate.toString() : "" , SamplingMode: mode ? mode.toString() : "" }, iotHubConnectionString);
+
+                this.outputLine(Constants.IoTHubDistributedTracingSettingLabel,
+                    `Update distributed tracing setting for device [${deviceIds.join(",")}] complete!` + (result ? result : ""));
+
+                if (node instanceof DistributedTracingLabelNode) {
+                    vscode.commands.executeCommand("azure-iot-toolkit.refresh", node);
+                } else if (node instanceof DistributedTracingSettingNode) {
+                    vscode.commands.executeCommand("azure-iot-toolkit.refresh", node.parent);
+                } else {
+                    vscode.commands.executeCommand("azure-iot-toolkit.refresh");
+                }
+            } catch (err) {
+                TelemetryClient.sendEvent(Constants.IoTHubAIUpdateDistributedSettingDoneEvent,
+                    { Result: "Fail", UpdateType: updateType.toString(), DeviceCount: deviceIds.length.toString(),
+                    SamplingRate: samplingRate ? "" : samplingRate.toString(), SamplingMode: mode ? "" : mode.toString() }, iotHubConnectionString);
+                this.outputLine(Constants.IoTHubDistributedTracingSettingLabel, `Failed to get or update distributed setting: ${err.message}`);
+                return;
+            }
+        });
+    }
+
+    private async updateDeviceTwin(enable: boolean, samplingRate: number, iotHubConnectionString: string, deviceIds: string[]) {
```

Comment:
Better to add annotation of return value for each function

Reply:
fixed